kernel/kobject.c

```diff
@@ -226,23 +227,44 @@ int kobject_lookup( struct kobject *kobject, const char *name, struct kobject **
 	return 0;
 }
 
-int kobject_copy( struct kobject *ksrc, struct kobject *kdst )
+int kobject_copy( struct kobject *ksrc, struct kobject **kdst )
```

Comment:
Suggest that you just return the new kobject as the return value.
Makes the function prettier to use below.